src/generic-bioengine-app.imjoy.html

```diff
@@ -249,7 +249,18 @@
         except Exception as exp:
             await api.alert(f"Failed to find model ({model_id}) in the BioEngine, error: {exp}")
             return
-        input_info = f", expected axes: {str(rdf['inputs'][0]['axes'])}, shape spec: {str(rdf['inputs'][0]['shape'])}"
+
+        # TODO the axis spec should be further simplified, e.g. we don't need the 'b' axis here, and if we have a single input channel we also don't need 'c'
```

Comment:
This is also pretty important to make this easier to understand for users, but it's a bit too complex to implement without any testing. But something like
```py
# this only works for the explicit shape spec, needs to be adapted to also work for the implicit case, where we have step and min
keep_axis = [i for i, (ax, sh) in enumerate(axis_spec, shape_spec) if (ax != "b" or not (ax == "c" and sh = 1))]
axis_spec = [axis_spec[i] for i in keep_axis]
shape_spec = [shape_spec[i] for i in keep_axis]
```

Reply:
Looks good, should we go for this now?

You can test it here too: https://imjoy.io/lite
Click the ImJoy icon on the upper-right corner, in the editor paste the source code and click Run.

Reply:
Sorry, I don't have any more time to work on this in the next two days. Please go ahead and implement and test this yourself.

Reply:
I need to sign off too, will skip this for now, and aim for another PR next time. Think a bit more, I think naive users can hardly understand even simpler case such as `axes: xyc`, maybe the best is to add a link for more detailed explanation in the future. Maybe something like "How to prepare your own data for test run?".

Reply:
>  Think a bit more, I think naive users can hardly understand even simpler case such as axes: xyc, maybe the best is to add a link for more detailed explanation in the future. Maybe something like "How to prepare your own data for test run?".

Yeah, I fully agree, we need to link to some more detailed explanation on how to prepare the data.
(But we should still make the message about axis and shapes here as simple as possible)

Reply:
cc @FynnBe, @esgomezm if you have some time in the next days that would be a worthwhile thing to do to significantly improve user experience

Reply:
Sorry, I'm not sure I understand what you mean here. Do you want further documentation in the BioImage.IO documentation about the test input and output tensors?

Reply:
> Do you want further documentation in the BioImage.IO documentation about the test input and output tensors?

No, we would want to add some documentation on how to use the "Test Run" button, including how to upload your own images so that they fit the expected tensor shapes of the model. There currently is a message like: `expected axes: bcyx, expected shape: minimum shape: [1, 1, 32, 32], valid shape increment: [0, 0, 16, 16]. For example the shape [1, 1, 64, 64] is valid)` for this, but it will not mean much to users that don't have any experience with the model spec.
(And we should also simplify this message further, I maybe have time to look into it today; but we'll need some extra info in a separate doc page in any case.)